src/averageAges.js

```diff
@@ -55,7 +67,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(child => child.sex === 'm' && people.find(
+      person => child.mother === person.name
+    )
+    )
+    : people.filter(child => people.find(
+      person => child.mother === person.name
+    )
+    );
```

Comment:
```suggestion
 const children = people.filter(child => people.find(
      person => child.mother === person.name));
  const NeededChildren = onlyWithSon
    ? children.filter(child => child.sex === 'm')
    : children;
```
I think you can make it easier...smth like this...not to repeat yourself) I have the same trouble with my solution before the student`s review)